libraries/AP_BattMonitor/AP_BattMonitor_Logging.cpp

```diff
@@ -7,6 +7,8 @@ extern const AP_HAL::HAL& hal;
 void AP_BattMonitor_Backend::Log_Write_BAT(const uint8_t instance, const uint64_t time_us) const
 {
     bool has_curr = has_current();
+    uint8_t percent = -1;
+    IGNORE_RETURN(capacity_remaining_pct(percent));
```

Comment:
This is a change in behaviour - we're now going to log 255 rather than zero here.

I'm actually OK with that change.

Reply:
Yes, I think there's more information in 255 than a zero that could be mistaked by an empty battery.